### PR TITLE
Added code to fix hardware reset issues 

### PR DESCRIPTION
**Pull request by jonlwowski012**
This code is to provide the updates to fix the bug where restarting a launch file with the realsense sensor often causes the cameras to fail.

I have not experienced the problem after this fix, but will test further to verify.

This fix came from the PR in the original intel repo here: https://github.com/intel-ros/realsense/pull/455/files

The original attempt to do this PR can be seen here. 
https://github.com/plusone-robotics/realsense/pull/15

### CODE REVIEW

**Comment by jonlwowski012** [comment]
@130s  I created a new PR for my convenience. This should hopefully be the final one.

**Comment by 130s** [comment]
Commit history looks clean and CI passed 👍 
Have you been able to confirm with the hardware this solves the issue?

**Comment by jonlwowski012** [comment]
Yes It does solve the issues

**Comment by 130s** [comment]
Apparently there are some tests in this repo, and on CI one of the tests failed but CI passed (this can happen with poor CI configuration, which for this repo was made in the upstream). I've looked back CI results from a few previous PRs, and I found the same was happening. This needs to be looked into.

The error seems to be triggered by trying to open non-existent file, which is not essential to the driver's code. However, this doesn't mean we can ignore the failing test. Because test fixture in this repo is made poorly, I can't yet tell whether there are more tests that are queued but didn't start because of failure or not.

https://travis-ci.org/plusone-robotics/realsense/builds/487481130#L3584
```
[ERROR] [1549033762.903179492]: An exception has been thrown: Failed to resolve request. Request to enable_device_from_file("/home/non_existent_file.txt") was invalid, Reason: Failed to create ros reader: Error opening file: /home/non_existent_file.txt
[FATAL] [1549033762.903710753]: Failed to load nodelet '/camera/realsense2_camera` of type `realsense2_camera/RealSenseNodeFactory` to manager `realsense2_camera_manager'
[camera/realsense2_camera-3] process has died [pid 17162, exit code 255, cmd /opt/ros/kinetic/lib/nodelet/nodelet load realsense2_camera/RealSenseNodeFactory realsense2_camera_manager __name:=realsense2_camera __log:=/home/travis/.ros/log/5ad5112a-2633-11e9-a276-42010a140003/camera-realsense2_camera-3.log].
log file: /home/travis/.ros/log/5ad5112a-2633-11e9-a276-42010a140003/camera-realsense2_camera-3*.log
[INFO] [1549033767.852109]: Un-Subscribing test colorStream
Killing ROS
[camera/realsense2_camera_manager-2] killing on exit
[ INFO] [1549033767.914212844]: Interrupt Signal is received! Terminating RealSense Node...
[camera/realsense2_camera_manager-2] escalating to SIGTERM
[rosout-1] killing on exit
[master] killing on exit
shutting down processing monitor...
... shutting down processing monitor complete
done
DONE
Test Failed: 'num_channels'
    TEST RESULTS    
--------------------
test name      score
---------  ---------
vis_avg_2         OK
vis_avg_1         OK
```

**Comment by jonlwowski012** [comment]
> Apparently there are some tests in this repo, and on CI one of the tests failed but CI passed (this can happen with poor CI configuration, which for this repo was made in the upstream). I've looked back CI results from a few previous PRs, and I found the same was happening. This needs to be looked into.
> 
> The error seems to be triggered by trying to open non-existent file, which is not essential to the driver's code. However, this doesn't mean we can ignore the failing test. Because test fixture in this repo is made poorly, I can't yet tell whether there are more tests that are queued but didn't start because of failure or not.
> 
> https://travis-ci.org/plusone-robotics/realsense/builds/487481130#L3584
> 
> ```
> [ERROR] [1549033762.903179492]: An exception has been thrown: Failed to resolve request. Request to enable_device_from_file("/home/non_existent_file.txt") was invalid, Reason: Failed to create ros reader: Error opening file: /home/non_existent_file.txt
> [FATAL] [1549033762.903710753]: Failed to load nodelet '/camera/realsense2_camera` of type `realsense2_camera/RealSenseNodeFactory` to manager `realsense2_camera_manager'
> [camera/realsense2_camera-3] process has died [pid 17162, exit code 255, cmd /opt/ros/kinetic/lib/nodelet/nodelet load realsense2_camera/RealSenseNodeFactory realsense2_camera_manager __name:=realsense2_camera __log:=/home/travis/.ros/log/5ad5112a-2633-11e9-a276-42010a140003/camera-realsense2_camera-3.log].
> log file: /home/travis/.ros/log/5ad5112a-2633-11e9-a276-42010a140003/camera-realsense2_camera-3*.log
> [INFO] [1549033767.852109]: Un-Subscribing test colorStream
> Killing ROS
> [camera/realsense2_camera_manager-2] killing on exit
> [ INFO] [1549033767.914212844]: Interrupt Signal is received! Terminating RealSense Node...
> [camera/realsense2_camera_manager-2] escalating to SIGTERM
> [rosout-1] killing on exit
> [master] killing on exit
> shutting down processing monitor...
> ... shutting down processing monitor complete
> done
> DONE
> Test Failed: 'num_channels'
>     TEST RESULTS    
> --------------------
> test name      score
> ---------  ---------
> vis_avg_2         OK
> vis_avg_1         OK
> ```

How do  we solve this then?

**Comment by 130s** [comment]
I can take a brief look into the tests and figure out what we need.

**Comment by jonlwowski012** [comment]
@130s when can  we get this merged?

**Review by 130s — APPROVED** [review]
https://github.com/plusone-robotics/realsense/pull/16#issuecomment-459791527 can be looked at separately (I'll open a ticket, maybe in the upstream) as it won't happen very soon, and physical test is passed.

File: .travis.yml [hunk 1]
```diff
@@ -3,7 +3,12 @@ dist: xenial
 
   # - git clone -v --progress https://github.com/doronhi/realsense.git  # This is Done automatically by TravisCI
 before_install:
-  - sudo ./.install_dependency.sh
```
Comment (130s) [inline .travis.yml:6]: My bad...I overlooked this undesired change.

Comment (130s) [reply .travis.yml:6]: Fix is opened https://github.com/plusone-robotics/realsense/pull/19

**Pull request merged**